web3/method.py

```diff
@@ -214,17 +214,25 @@ def process_params(self, module: "Module", *args: Any, **kwargs: Any) -> Tuple[
 
 class DeprecatedMethod:
     def __init__(
-        self, method: Method[Callable[..., Any]], old_name: str, new_name: str
+        self,
+        method: Method[Callable[..., Any]],
+        old_name: str,
+        new_name: Optional[str] = None,
+        msg: Optional[str] = None,
     ) -> None:
         self.method = method
         self.old_name = old_name
         self.new_name = new_name
+        self.msg = msg
 
     def __get__(
         self, obj: Optional["Module"] = None, obj_type: Optional[Type["Module"]] = None
     ) -> Any:
-        warnings.warn(
-            f"{self.old_name} is deprecated in favor of {self.new_name}",
-            category=DeprecationWarning,
-        )
+        if self.msg is not None:
+            warnings.warn(self.msg)
```

Comment:
@kclowes should this include the `category=DeprecationWarning` argument as well?